Framework/Core/src/DataRelayer.cxx

```diff
@@ -7,6 +7,7 @@
 // In applying this license CERN does not waive the privileges and immunities
 // granted to it by virtue of its status as an Intergovernmental Organization
 // or submit itself to any jurisdiction.
+#include "Framework/RootSerializationSupport.h"
```

Comment:
I think it's fine to have the `.cxx` know about ROOT. In any case there will be someone using ROOT across the whole topology and the library will have to be linked in at least once. The purpose of this is to avoid exposing the header dependency, which hopefully will reduce compilation complexity for those who do not need ROOT. 
If we have one full topology without ROOT, we can of course separate this further.

Notice that for now I've also kept some sort of backward compatibility so that we do not modify (for now) 100s of header files which relied on DataAllocator.h exposing ROOT headers.